doc/manipulating_images/input_output.rst

```diff
@@ -0,0 +1,258 @@
+.. _extracting_data:
+
+=====================================================
+Input and output: neuroimaging data representation
+=====================================================
+
+.. contents:: **Contents**
+    :local:
+    :depth: 1
+
+|
+
+.. currentmodule:: nilearn.image
+
+.. _loading_data:
+
+Inputing data: file names or image objects
+===========================================
+
+File names and objects, 3D and 4D images
+-----------------------------------------
+
+All Nilearn functions accept file names as arguments::
+
+    >>> from nilearn import image
+    >>> smoothed_img = image.smooth_img('/home/user/t_map001.nii')  # doctest: +SKIP
+
+Nilearn can operate on either file names or `NiftiImage objects
+<http://nipy.org/nibabel/nibabel_images.html>`_. The later represent the
+data loaded in memory. In the example above, the
+function :func:`smooth_img` returns a NiftiImage object, which can then
```

Comment:
Technically a Nifti1Image. I know that it may sound weird for the user, but he may be even more puzzled by if he sees that his type (Nifti1Image) does not match the one in the doc.